pyarr/radarr_api.py

```diff
@@ -0,0 +1,998 @@
+        """
+        params = {
+            "page": page,
+            "pageSize": page_size,
+            "sortDirection": sort_direction,
+            "sortKey": sort_key,
+        }
+        path = "/api/v3/history"
+        res = self.request_get(path, params=params)
+        return res
+
+    # GET /history/movie
+    def get_movie_history(self, id_, event_type=None):
+        """Return a json object list of items in your history.
+
+        Args:
+            [Required] id_ (int) - Database ID of movie
+            [Optional] event_type (int) - History event type to retrieve
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"movieId": id_}
+        if event_type:
+            params["eventType"] = event_type
+        path = "/api/v3/history/movie"
+        res = self.request_get(path, params=params)
+        return res
+
+    ## BLACKLIST
+
+    # GET /blacklist
+    def get_blacklist(
+        self,
+        page,
+        page_size=20,
+        sort_direction="descending",
+        sort_key="date",
+    ):
+        """Returns blacklisted releases.
+        Args:
+            [Required] page (int)
+            [Optional] page_size (int) - Default: 20
+            [Optional] sort_direction (str) - Default: descending
+            [Optional] sort_key (str) - Default: date
+        Returns:
+            JSON Response (dict)
+        """
+        params = {
+            "page": page,
+            "pageSize": page_size,
+            "sortDirection": sort_direction,
+            "sortKey": sort_key,
+        }
+        path = "/api/v3/blacklist"
+        res = self.request_get(path, params=params)
+        return res
+
+    # DELETE /blacklist
+    def del_blacklist(self, id_):
+        """Removes a specific release (the id provided) from the blacklist.
+        Args:
+            [Required] id_ (int)
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"id": id_}
+        path = "/api/v3/blacklist"
+        res = self.request_del(path, params=params)
+        return res
+
+    # GET /blacklist/movie
+    def get_blacklist_by_movie_id(
+        self,
+        id_,
+    ):
+        """Retrieves blacklisted releases that are tied to a given movie in the database.
+        Args:
+            [Required] id_ (int)
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"movieId": id_}
+        path = "/api/v3/blacklist/movie"
+        res = self.request_get(path, params=params)
+        return res
+
+    # DELETE /blacklist/bulk
+    def del_blacklist_bulk(self, data):
+        """Delete blacklisted releases in bulk.
+        Args:
+            [Required] data (dict)
+        Returns:
+            JSON Response (dict)
+        """
+        path = "/api/v3/blacklist/movie"
+        res = self.request_del(path, data=data)
+        return res
+
+    ## QUEUE
+
+    # GET /queue
+    def get_queue(
+        self,
+        page,
+        page_size=20,
```

Comment:
set defaults `page=1, page_size=10`

Reply:
For now I have left page_size = 20 to follow Radarr API defaults on these endpoints. 

Let's discuss whether we should follow the the Radarr docs, or match Sonarr docs.